receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -315,16 +309,12 @@ func (c *kafkaMetricsConsumer) Start(_ context.Context, host component.Host) err
 			headers: c.headers,
 		}
 	}
-	go func() {
-		if err := c.consumeLoop(ctx, metricsConsumerGroup); err != nil {
-			componentstatus.ReportStatus(host, componentstatus.NewFatalErrorEvent(err))
-		}
-	}()
+	go c.consumeLoop(ctx, metricsConsumerGroup)
```

Comment:
I am a little nervous that this is creating a routine with no wait means that it could leak routines on shutdown, would this be something you're able to address within this PR?

Reply:
No problem!